### PR TITLE
add argument spec and automated docs

### PR DESCRIPTION
**Pull request by rndmh3ro**
This PR adds several things:

- an argument spec (https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html#role-argument-validation) to be used to validate if your role has the required params
- building upon the argument spec:
  - an automatically created and updated readme (with https://github.com/telekom-mms/Automated-Ansible-Role-Documentation) 
    - the readme vars are created from the argument_spec.yml, so every new default-variable has to be added there to land in the readme 
    - the format of the readme can be adjusted (e.g. if you want to keep the table format)
  - a github action to automatically update the readme

Let me know what you think

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,82 +1,132 @@
...
+
+## Role Variables
+
+- `haproxy_backend_vars`
+  - Default: ``
+  - Description: List of backend variables
+  - Type: list
+  - Required: no
+- `haproxy_backends`
+  - Default: `[{"name": null, "default": "myHttpBackend", "type": "str", "description": "", "connect_timeout": {"default": "10s", "type": "str", "description": ""}, "server_timeout": {"default": "1m", "type": "str", "description": ""}, "mode": {"default": "http", "type": "str", "description": ""}, "servers": [{"name": null, "default": "my-server-1", "type": "str", "description": "", "address": {"default": "192.168.0.3", "type": "str", "description": ""}, "port": {"default": 80, "type": "str", "description": ""}}, {"name": null, "default": "my-server-2", "type": "str", "description": "", "address": {"default": "192.168.0.3", "type": "str", "description": ""}, "port": {"default": 80, "type": "str", "description": ""}}], "options": ["httplog", "http-check"]}]`
```
[the remark this answers is not in the document]
Comment (rndmh3ro) [reply README.md:21]: No, not easily...
Thinking about this some more, is such a default even useful?

[the remark this answers is not in the document]
Comment (rndmh3ro) [reply README.md:21]: @beechesII, what do you think? Should we remove the defaults for the backends and frontends?

[the remark this answers is not in the document]
Comment (beechesII) [reply README.md:21]: schurzi is right, they are more like an example than a default. You can remove them from defaults file.

[the remark this answers is not in the document]
Comment (rndmh3ro) [reply README.md]: done

**Review by beechesII — APPROVED** [review]
lgtm, thx for implementation

**Pull request merged**